lib/storage/src/content_manager/toc.rs

```diff
@@ -299,6 +303,11 @@ impl TableOfContent {
         operation: CreateCollection,
         collection_shard_distribution: CollectionShardDistribution,
     ) -> Result<bool, StorageError> {
+        // Collection operations require multiple file operations,
+        // before collection can actually be registered in the service.
+        // To prevent parallel writing of the files, we use this lock.
+        let _lock = self.collection_create_lock.lock().await;
```

Comment:
I think we can drop `_lock` earlier, right after inserting it in the list of collections.

Or do we keep the lock for the lifetime of the whole function on purpose? That includes keeping it while running our `init_from` logic.

Reply:
agreed